client/src/pages/Search.js

```diff
@@ -10,9 +10,12 @@ function Search() {
 
     const [loading, setLoading] = useState(false)
 
+    const [searchPerformed, setSearchPerformed] = useState(false)
+
     const handleSearch = async (query) => {
```

Comment:
Can we wrap this in a `useCallback()` hook?

It's getting passed to the child component and we don't want a bunch of unnecessary re-renders in the asynchronous function.

Maybe something like:

```ts
const handleSearch = useCallback(async (query) => {
    setLoading(true);
    setSearchPerformed(true);

    try {
      const response = await fetch(`${HOST}/api/pipeline/search/${query}`, {
        method: 'GET',
        headers: {
          'Content-Type': 'application/json',
        },
      });

      if (!response.ok) {
        throw new Error(`HTTP error! Status: ${response.status}`);
      }

      const data = await response.json();
      setProfiles([...data]);
      setLoading(false);
    } catch (error) {
      console.error(error.message);
      setLoading(false);
    }
  }, []);
  ```
  
  And then inside the `<QuerySearchInput />` component

```ts
....
const handleInputChange = useCallback((event) => {
    const inputValue = event.target.value;
    setQuery(inputValue);

    if (inputValue.length > 0) {
      const filteredCompanies = companies.filter((company) =>
        company.name.toLowerCase().startsWith(inputValue.toLowerCase())
      );
      setResults(filteredCompanies);
    } else {
      setResults([]);
    }
  }, []);

  const handleCompanyButtonClick = async (company) => {
    setQuery('');
    setResults([]);

    await handleSearch(company.name);
  };
....
```

This could be resolved in a future PR, so fine by me if we **don't**, but just calling it out.